libsawtooth/src/receipt/store/mod.rs

```diff
@@ -87,3 +92,51 @@ pub trait ReceiptStore: Sync + Send {
         id: Option<String>,
     ) -> Result<Box<dyn Iterator<Item = TransactionReceipt>>, ReceiptStoreError>;
 }
+
+#[cfg(feature = "transaction-receipt-store-lmdb")]
+impl AsBytes for TransactionReceipt {
+    fn as_bytes(&self) -> Vec<u8> {
+        self.clone()
+            .into_bytes()
+            .expect("failed to serialize transaction receipt")
+    }
+}
+
+#[cfg(feature = "transaction-receipt-store-lmdb")]
+impl FromBytes for TransactionReceipt {
+    fn from_bytes(bytes: &[u8]) -> Result<Self, String> {
+        <TransactionReceipt as ReceiptFromBytes<TransactionReceipt>>::from_bytes(bytes)
+            .map_err(|err| err.to_string())
+    }
+}
+
+/// Trait used by the LMDB `ReceiptStore` implementation to convert an index to bytes
+/// for storage.
+#[cfg(feature = "transaction-receipt-store-lmdb")]
+pub trait AsBytes {
```

Comment:
Suggest trying to avoid defining a new AsBytes trait here if possible.

Reply:
Removed this AsBytes trait definition

Reply:
Removed FromBytes as well